components/ActionBox/components/DealActionBox.tsx

```diff
@@ -0,0 +1,265 @@
+import Image from 'next/image';
+import styled from 'styled-components';
+import { useEffect, useState, useMemo } from 'react';
+
+import Info from 'assets/svg/info.svg';
+
+import Connector from 'containers/Connector';
+
+import { Status } from 'components/DealStatus';
+import QuestionMark from 'components/QuestionMark';
+import ConfirmTransactionModal from 'components/ConfirmTransactionModal';
+
+import { FlexDivRowCentered, Tooltip } from '../../common';
+
+import {
+	Container,
+	ErrorNote,
+	ContentContainer,
+	ActionBoxInputLabel,
+	InputContainer,
+	ActionBoxInput,
+	ActionBoxMax,
+	ActionBoxHeaderWrapper,
+	ActionBoxHeader,
+	FlexDivCenterRow,
+	ActionButton,
+} from './commons';
+
+import { TransactionType, TransactionStatus } from 'constants/transactions';
+import { statusToText, swimmingPoolID } from 'constants/pool';
+
+const DealActionBox = ({
+	poolId,
+	dealRedemptionData,
+	purchaseCurrency,
+	transaction: { txType, txState, setTxType, setGasPrice, gasLimitEstimate, onSubmit },
+	input: { placeholder, label, maxValue, inputValue, setInputValue, setIsMaxValue },
+}: any) => {
+	const { walletAddress } = Connector.useContainer();
+	const [showTooltip, setShowTooltip] = useState(false);
+	const [showTxModal, setShowTxModal] = useState(false);
+	const [isDealAccept, setIsDealAccept] = useState(true);
+
+	const isWithdraw = !isDealAccept;
+
+	const isPoolDisabled = [swimmingPoolID].includes(poolId);
+
+	const isDisabled: boolean = useMemo(
+		() => !walletAddress || !isWithdraw || isPoolDisabled,
+		[walletAddress, isWithdraw, isPoolDisabled]
+	);
+
+	useEffect(() => {
```

Comment:
We need to find a better way. I still consider this as a hack. Open question: do we even need TransactionStatus? I'm not sure.